"""Укажите причину, по которой в результате работы данного кода ничего не было выведено в консоль:"""


class TrickyManager:
    def __init__(self, lst):
        self.worklist = lst

    def __enter__(self):
        __builtins__.print = lambda x, f=open('trace.log', 'a'), \
                                          end='\n': f.write(str(x))
        return 'T-R-I-C-K-Y'

    def __exit__(self, exc_type, exc_value, trace):
        return True


with TrickyManager([1,2,3,4,5]) as t:
    for c in t:
        print(c, end=' ')

print(t)



# =================================================================================================
# 1) Данный код - нерабочий.
# 2) Метод __exit__ не должен возвращать True, так как исключение не было обработано внутри метода.
# 3) Метод __enter__ не возвращает объект self.
# 4) В методе __exit__ не происходит восстановление контекста и освобождение ресурсов. * Или это
# 5) Объект t - пустой объект, поэтому нечего выводить в консоль.
# 6) Список, передаваемый в конструктор объекта TrickyManager, никак не используется. * Или это
